### PR TITLE
Fixed open/closed interval definitions

### LINKED ISSUE #231: Intervals Closed/Open backwards??
Hi,

reading the docs and [the code](https://github.com/antonagestam/phantom-types/blob/main/src/phantom/predicates/interval.py), I feel there's an inversion.

```python
def open(low: T, high: T) -> Predicate[SupportsLeGe[T]]:
    """
    Create a predicate that succeeds when its argument is in the range ``(low, high)``.
    """

    @bind_name(open, low, high)
    def check(value: SupportsLeGe[T]) -> bool:
        return low <= value <= high

    return check
```
Here I would have expected `return low < value < high`

https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology

In an open interval, bounds are not included, while in a closed one, they are.

Am I missing something? Am I reading the code wrong?

### PR DESCRIPTION
**Pull request by g-as**
Fixes #231

### CODE REVIEW

**Comment by antonagestam** [comment]
Hi @g-as, thanks for putting in the time to address is!

I'm again pondering the issue of backwards compatibility. I'm now thinking it would make sense to rename the predicate functions and the types to use terms "inclusive" and "exclusive". In my opinion those names are more intuitive, but it would also solve the issue of backwards compatibility.

Just to clarify a bit about my reasoning here, I'll expand a little bit. In releasing a version `1.0`, I'm not too worried about removing a function or a class. If someone fails to read the release notes, it's still a change that should be easily discovered by a test suite and probably in 99% of usages this will break at import-time. OTH, renaming `closed -> open` and vice versa requires a much more disciplined test suite to discover and I'm worried it will sail under the radar into production code before discovered.

What do you think? If you agree it's a good idea, please feel free to go ahead and make those changes on this PR.

Thanks again for the work thus far! 🙏

**Comment by g-as** [comment]
Agreed that breakage is the best way to avoid behaviour change without the user's knowledge. So renaming with appropriate names, and arguably better/more explicit names is the best way to go.

Changes incoming asap.

File: src/phantom/predicates/interval.py [hunk 1]
```diff
@@ -17,49 +17,49 @@
 T = TypeVar("T")
 
 
-def open(low: T, high: T) -> Predicate[SupportsLeGe[T]]:
+def open(low: T, high: T) -> Predicate[SupportsLtGt[T]]:
```
Comment (antonagestam) [inline src/phantom/predicates/interval.py:20]: I believe we should rename `open` and `closed` in this module too, as they are part of the public API of the library.

**Comment by antonagestam** [comment]
Thank you so much for the help, and again for making aware of this 🍰

**Comment by g-as** [comment]
My pleasure.

Keep up the good work.

**Pull request merged**